File: 0020-valid-parentheses/0020-valid-parentheses.py
class Solution:
    def isValid(self, s: str) -> bool:
        stack = []
        for st in s:
            if st == '(' or st == '{' or st == '[':
                stack.append(st)
            else:
                if len(stack) == 0:
                    return False
                else:
                    if (stack[len(stack)-1] == '(' and st == ')') or (stack[len(stack)-1] == '{' and st == '}') or (stack[len(stack)-1] == '[' and st == ']'):
                        stack.pop()
                    else:
                        return False
        if len(stack) != 0:
            return False
        return True

File: 0020-valid-parentheses/0020-valid-parentheses.cpp
class Solution {
public:
    bool isValid(string s) {
        stack<char> st;
        for(int i=0; i<s.length(); i++){
            if(s[i]=='(' || s[i]=='{' || s[i]=='[') st.push(s[i]);
            else{
                if(st.empty()) return false;
                if(s[i]==')'){
                    if(st.top()!='(') return false;
                    st.pop();
                }
                else if(s[i]=='}'){
                    if(st.top()!='{') return false;
                    st.pop();
                }
                else{
                    if(st.top()!='[') return false;
                    st.pop();
                }
            }
        }
        if(!st.empty()) return false;
        return true;
    }
};